app/javascript/utilities/gist.js

```diff
@@ -30,11 +30,13 @@ function getGistTags(nodes) {
 
 function loadEmbeddedGists(postscribe, gistTags) {
   for (const gistTag of gistTags) {
-    postscribe(gistTag, gistTag.firstElementChild.outerHTML, {
-      beforeWrite(text) {
-        return gistTag.childElementCount > 3 ? '' : text;
-      },
-    });
+    setTimeout(() => {
+      postscribe(gistTag, gistTag.firstElementChild.outerHTML, {
+        beforeWrite(text) {
+          return gistTag.childElementCount > 3 ? '' : text;
+        },
+      });
+    }, 500);
```

Comment:
Instead of adding a fixed timeout to each call, is it possible to instead `await` the postscribe function so each update runs sync (reading through the various github issue history it seems that's the problem)?

```suggestion
    await postscribe(gistTag, gistTag.firstElementChild.outerHTML, {
      beforeWrite(text) {
        return gistTag.childElementCount > 3 ? '' : text;
      },
    });
```

Reply:
Hi, this didnt work. First I tried , since using `await` directly is not possible
```

async function loadEmbeddedGists(postscribe, gistTags) {
    for (const gistTag of gistTags) {
      await postscribe(gistTag, gistTag.firstElementChild.outerHTML, {
          beforeWrite(text) {
            return gistTag.childElementCount > 3 ? '' : text;
          }
        });
    }
}
```
This didnt solve the problem.

Using async  inside the function resulted in `Failed to execute 'write' on 'Document': It isn't possible to write into a document from an asynchronously-loaded external script unless it is explicitly opened.`

BTW, the (documentation)[https://github.com/krux/postscribe#getting-started] suggests `It is best and safest to use PostScribe after DOM is ready.`. I think code is not working that way, but maybe I'm wrong


There was also another way I found to be working, though I don't exactly know why it did. When loading gists from backend instead of returning `<script>` tags I return `<span>`. Then in the `gist.js` file 
```
postscribe(gistTag, gistTag.firstElementChild.outerHTML.replaceAll("span", "script"), {
          beforeWrite(text) {
            return gistTag.childElementCount > 3 ? '' : text;
          }
        });
```        
I think this works because the documentation suggests we build script tags dynamically.

Reply:
> There was also another way I found to be working, though I don't exactly know why it did. When loading gists from backend instead of returning <script> tags I return <span>.

This is interesting! Unfortunately I don't think changing the HTML of the gist tag is the best idea, because AFAIK comments for example are stored in their HTML entirety, so older comments with gists would encounter the same problem.

Your finding did make me notice something too - after we execute `postScribe` we end up with _two_ script tags in each gist tag, e.g.

![CleanShot 2021-09-10 at 13 48 36](https://user-images.githubusercontent.com/20773163/132855628-b400c58c-e6fc-456f-84fa-33f2146ef31e.png)

I experimented with removing the original script tag after we call postscribe, and it seems to resolve the issue for me:

```js
function loadEmbeddedGists(postscribe, gistTags) {
  for (const gistTag of gistTags) {
    const gistWrapper = gistTag.firstElementChild;
    postscribe(gistTag, gistWrapper.outerHTML, {
      beforeWrite(text) {
        return gistTag.childElementCount > 3 ? '' : text;
      },
    });
    gistWrapper.remove();
  }
}
```

I'm still trying to unpick _exactly_ what's going wrong when we have those two script tags, but it seems like we definitely don't want two anyway 🙃  (CC @nickytonline)

Reply:
are we planning to ahead with this?